server.js

```diff
@@ -0,0 +1,90 @@
+const
+restify   = require('restify'),
+mongoose  = require('mongoose'),
+
+userRoute = require('./routes/userRoute'),
+config    = require('./modules/config'),
+logger    = require('./modules/logger'),
+
+dbUri     = config.getDbConnectionString();
+
+function startServer(port = 8080) {
+    const uri    = 'localhost:' + port,
+          server = restify.createServer({
+              name: 'wirecraft-test-api',
+              version: '1.0.0'
+          });
+
+    server.use(restify.acceptParser(server.acceptable));
+    server.use(restify.queryParser());
+    server.use(restify.bodyParser());
+    server.use(restify.CORS());
+
+    server.listen(port);
+    logger.info('Server address:', server.address());
+
+    return { server, uri };
+}
+
+function startDatabase(uri) {
+    mongoose.Promise = Promise;
+    mongoose.connect(uri);
+
+    const db = mongoose.connection;
+    db.on('error', function (err) {
+        logger.error('connection error:', err.message);
+    });
+    db.once('open', () => {
+        logger.info('connected to mongo');
+    });
+
+    return () => { mongoose.disconnect(); };
+}
+
+function processRouteFor(url) {
+    return (route) => {
+        return (req, res, next) => {
+
+            logger.trace('REQUEST:', req.params);
+
+            route(req.params, (err, body, options = {}) => {
+                const status = options.status || 200;
+                if (err) {
+                    logger.error(err);
+                    res.send(400, err.message || 'could not process request');
+                } else {
+                    logger.trace('RESPONSE:', status, body);
+
+                    if (options.location) {
+                        res.setHeader('Location', url + options.location);
+                    }
+
+                    res.send(status, body);
+                }
+
+                next();
```

Comment:
Because of the restify library, it allows for multiple handlers for the same route and `next()` calls the next handler. You could add a general handler for logging, or for handling authentication headers for example.